video/out/opengl/egl_helpers.c

```diff
@@ -120,22 +147,44 @@ static bool create_context(struct ra_ctx *ctx, EGLDisplay display,
         name = "GLES 2.x +";
     }
 
+    const char *egl_exts = eglQueryString(display, EGL_EXTENSIONS);
+
     MP_VERBOSE(ctx, "Trying to create %s context.\n", name);
 
     if (!eglBindAPI(api)) {
         MP_VERBOSE(ctx, "Could not bind API!\n");
         return false;
     }
 
+    bool request_float_fmt = opts->output_format < 0;
+    int depth_packed = request_float_fmt ? -opts->output_format : opts->output_format;
+    EGLint r_size = depth_packed >> 24,
+           g_size = (depth_packed >> 16) & 0xff,
+           b_size = (depth_packed >> 8) & 0xff,
+           a_size = depth_packed & 0xff;
+    bool has_float_format_ext = gl_check_extension(egl_exts, "EGL_EXT_pixel_format_float");
+    if (request_float_fmt && !has_float_format_ext) {
+        MP_MSG(ctx, msgl, "Could not request floating point pixel format for %s!\n", name);
+        return false;
+    }
+
     EGLint attributes[] = {
         EGL_SURFACE_TYPE, EGL_WINDOW_BIT,
-        EGL_RED_SIZE, 8,
-        EGL_GREEN_SIZE, 8,
-        EGL_BLUE_SIZE, 8,
-        EGL_ALPHA_SIZE, ctx->opts.want_alpha ? 8 : 0,
         EGL_RENDERABLE_TYPE, rend,
+        EGL_RED_SIZE, MPMAX(r_size, 8),
+        EGL_GREEN_SIZE, MPMAX(g_size, 8),
+        EGL_BLUE_SIZE, MPMAX(b_size, 8),
+        EGL_ALPHA_SIZE, opts->output_format ? a_size : (ctx->opts.want_alpha ? 8 : 0),
+        opts->output_format && has_float_format_ext ? EGL_COLOR_COMPONENT_TYPE_EXT : EGL_NONE,
+        request_float_fmt ? EGL_COLOR_COMPONENT_TYPE_FLOAT_EXT : EGL_COLOR_COMPONENT_TYPE_FIXED_EXT,
```

Comment:
you might need to add a fallback define for these at the top in case the system egl is older

Reply:
fixed